lib/components/base/Pagination.vue

```diff
@@ -42,73 +42,63 @@
       :tabindex="page === pages[pages.length - 1] ? -1 : 0"
       class="right-arrow paginate has-icon"
       aria-label="Next Page"
-      :href="linkFunction(page + 1)"
+      :href="linkFunction?.(page + 1)"
       @click.prevent="page !== pages[pages.length - 1] ? switchPage(page + 1) : null"
     >
       <RightArrowIcon />
     </a>
   </div>
 </template>
-<script setup>
+<script setup lang="ts">
+import { computed } from 'vue'
 import { GapIcon, LeftArrowIcon, RightArrowIcon } from '@'
-</script>
-<script>
-import { defineComponent } from 'vue'
-
-export default defineComponent({
-  props: {
-    page: {
-      type: Number,
-      default: 1,
-    },
-    count: {
-      type: Number,
-      default: 1,
-    },
-    linkFunction: {
-      type: Function,
-      default() {
-        return null
-      },
-    },
-  },
-  emits: ['switch-page'],
-  computed: {
-    pages() {
-      let pages = []
-
-      if (this.count > 7) {
-        if (this.page + 3 >= this.count) {
-          pages = [
-            1,
-            '-',
-            this.count - 4,
-            this.count - 3,
-            this.count - 2,
-            this.count - 1,
-            this.count,
-          ]
-        } else if (this.page > 5) {
-          pages = [1, '-', this.page - 1, this.page, this.page + 1, '-', this.count]
-        } else {
-          pages = [1, 2, 3, 4, 5, '-', this.count]
-        }
-      } else {
-        pages = Array.from({ length: this.count }, (_, i) => i + 1)
-      }
-
-      return pages
-    },
-  },
-  methods: {
-    switchPage(newPage) {
-      this.$emit('switch-page', newPage)
-      if (newPage !== null && newPage !== '' && !isNaN(newPage)) {
-        this.$emit('switch-page', Math.min(Math.max(newPage, 1), this.count))
-      }
-    },
-  },
+
+const emit = defineEmits<{
+  'switch-page': [page: number]
+}>()
+
+const props = withDefaults(
+  defineProps<{
+    page: number
+    count: number
+    linkFunction: (page: number) => string | undefined
+  }>(),
+  {
+    page: 1,
+    count: 1,
+    linkFunction: () => undefined,
```

Comment:
```suggestion
    linkFunction: (page: number) => void page,
```

It seems that Volar is pretty stupid and thinks that this function is an initialiser that returns `undefined`, so it marks `linkFunction` as `((page: number) => string | undefined) | undefined` (essentially making it optional, even though it isn't, because that's not an initialiser!!). However, matching the original signature by adding an argument seems to fix this. Weird and is probably worth reporting to Volar...

Anyway, do this and then you can avoid using optional chaining for every `linkFunction` call.